src/main/java/org/mitre/synthea/export/rif/DMEExporter.java

```diff
@@ -0,0 +1,210 @@
+package org.mitre.synthea.export.rif;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.List;
+import org.mitre.synthea.export.ExportHelper;
+import org.mitre.synthea.world.agents.Clinician;
+import org.mitre.synthea.world.agents.Person;
+import org.mitre.synthea.world.concepts.Claim;
+import org.mitre.synthea.world.concepts.ClinicianSpecialty;
+import org.mitre.synthea.world.concepts.HealthRecord;
+
+/**
+ * Exporter for RIF DME (durable medical equipment) file.
+ */
+public class DMEExporter extends RIFExporter {
+
+  /**
+   * Construct an exporter for DME claims.
+   * @param startTime earliest claim date to export
+   * @param stopTime end time of simulation
+   * @param exporter the exporter instance that will be used to access code mappers
+   */
+  public DMEExporter(long startTime, long stopTime, BB2RIFExporter exporter) {
+    super(startTime, stopTime, exporter);
+  }
+
+  /**
+   * Export DME details for a single person.
+   * @param person the person to export
+   * @return count of claims exported
+   * @throws IOException if something goes wrong
+   */
+  long export(Person person) throws IOException {
+    long claimCount = 0;
+    HashMap<BB2RIFStructure.DME, String> fieldValues = new HashMap<>();
+
+    for (HealthRecord.Encounter encounter : person.record.encounters) {
+      if (encounter.stop < startTime || encounter.stop < CLAIM_CUTOFF) {
+        continue;
+      }
+      if (RIFExporter.isVAorIHS(encounter)) {
+        continue;
+      }
+
+      long claimId = RIFExporter.nextClaimId.getAndDecrement();
+      long claimGroupId = RIFExporter.nextClaimGroupId.getAndDecrement();
+      long carrClmId = RIFExporter.nextCarrClmCntlNum.getAndDecrement();
+
+      double latestHemoglobin = 0;
+      for (HealthRecord.Observation observation : encounter.observations) {
+        if (observation.containsCode("718-7", "http://loinc.org")) {
+          latestHemoglobin = (double) observation.value;
+        }
+      }
+
+      fieldValues.clear();
+      exporter.staticFieldConfig.setValues(fieldValues, BB2RIFStructure.DME.class, person);
+
+      // complex fields that could not easily be set using cms_field_values.tsv
+      fieldValues.put(BB2RIFStructure.DME.CLM_ID, "" + claimId);
+      fieldValues.put(BB2RIFStructure.DME.CLM_GRP_ID, "" + claimGroupId);
+      fieldValues.put(BB2RIFStructure.DME.CARR_CLM_CNTL_NUM, "" + carrClmId);
```

Comment:
Looking at:

```java
    long carrClmId = RIFExporter.nextCarrClmCntlNum.getAndDecrement();
```
and then

```java
     fieldValues.put(BB2RIFStructure.DME.CARR_CLM_CNTL_NUM, "" + carrClmId);
```

I wondered if we're doing this correctly... so I looked up `BB2RIFStructure.DME.CARR_CLM_CNTL_NUM` and couldn't find a definition for it anywhere.

Reply:
I'm wondering if this is supposed to reference an actual carrier claim, meaning whether or not it should have actual referential integrity (which this approach will not satisfy).

Reply:
I agree that we should look into this but it seems out of scope for this PR since this functionality has not been changed.